scylla/src/transport/session.rs

```diff
@@ -2117,9 +2108,8 @@ impl RequestSpan {
         }
     }
 
-    pub(crate) fn new_prepared(
-        prepared_metadata: &PreparedMetadata,
-        partition_key: Option<&Bytes>,
+    pub(crate) fn new_prepared<'ps>(
+        partition_key: Option<impl Iterator<Item = (&'ps [u8], &'ps ColumnSpec)> + Clone>,
```

Comment:
Does this need to be that much generic? Does it really make sense to put here anything that isn't an `Option<PartitionKey>`? Perhaps doing so will make it easier to fix the lifetime issue in `partition_key_displayer`.

Reply:
I've tried it up and gave up. Apparently, the `Iterator` returned by `PartitionKey::iter()` is less problematic in terms of lifetime issue than `PartitionKey` itself.